client/web/compose/src/views/Admin/Pages/Builder.vue

```diff
@@ -462,6 +493,73 @@ export default {
 
       return true
     },
+
+    cloneBlock (index) {
+      const blockClone = Object.assign({}, this.blocks[index])
+      this.appendBlock(blockClone, this.$t('notification:page.cloneSuccess'))
+    },
+
+    async copyBlock (index) {
+      const parsedBlock = JSON.stringify(this.blocks[index])
+      navigator.clipboard.writeText(parsedBlock).then(() => {
+        this.toastSuccess(this.$t('notification:page.copySuccess'))
+        this.toastInfo(this.$t('notification:page.blockWaiting'))
+      },
+      (err) => {
+        this.toastErrorHandler(this.$t('notification:page.copyFailed', { reason: err }))
+      })
+    },
+
+    pasteBlock (event) {
+      if (this.$route.name === 'admin.pages.builder') {
+        const active = document.activeElement
+        const inputs = document.querySelectorAll('input')
+        let inputActive = false
+
+        // Not to intrude on paste for any input field
+        // Using forEach because other array methods don't work on NodeList
+        inputs.forEach(input => {
+          if (input === active) {
+            inputActive = true
+          }
+        })
+
+        if (!inputActive) {
+          event.preventDefault()
+          const paste = (event.clipboardData || window.clipboardData).getData('text')
+          // Doing this to handle JSON parse error
+          try {
+            const block = JSON.parse(paste)
+            const valid = this.isValid(block)
+
+            if (valid) {
+              this.appendBlock(block, this.$t('notification:page.pasteSuccess'))
+            }
+          } catch (error) {
+            this.toastWarning(this.$t('notification:page.invalidBlock'))
+            console.log(error)
+          }
+        }
+      }
+    },
+
+    appendBlock (block, msg) {
+      if (this.blocks.length) {
+        // ensuring we append the block to the end of the page
+        // eslint-disable-next-line
+        const [, y, , ,] = this.blocks[this.blocks.length - 1].xywh
```

Comment:
This won't work since there is no guarantee that the last block is the one at the most bottom. Find the max y of this.blocks and set it to that